llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -231,6 +231,53 @@ class VPLane {
   }
 };
 
+class VPInstruction;
+class VPCSAHeaderPHIRecipe;
+class VPCSADataUpdateRecipe;
+class VPCSAExtractScalarRecipe;
+
+/// VPCSAState holds information required to vectorize a conditional scalar
+/// assignment.
+class VPCSAState {
+  VPValue *VPInitScalar = nullptr;
```

Comment:
Would be  good to avoid holding references to VPValues/Recipes, as they may be removed/replaced, in case the state here gets stale.

Reply:
removed